### ISSUE TITLE
Add example of how to test analysis

### ISSUE THREAD

**Issue by Danack** [psalm/psalm-plugin-skeleton#2]
So thanks for setting this up.

Please could you either add an example of how to test plugin detects errors correctly, or just tell me how to test [some code](https://github.com/Danack/PropertyTaint/blob/33bc2369a9832a0c646df169ffdeb827c15185ff/tests/PluginTest.php#L57-L60)

I can see in Psalm itself, that code is checked with [this](
[https://github.com/vimeo/psalm/blob/56cddd16bf7c6badd8d6d2b44f427f18e08b40de/tests/TaintTest.php#L38-L66] but that seems to require a lot of infrastructure code that isn't shipped with Psalm.

So I'm kind of lost on how to test a plugin behaves correctly.

**Comment by weirdan** [comment]
Many of the plugins are tested with Codeception + `weirdan/codeception-psalm-module` package.

1. first you would need to set up Codeception itself as well as the psalm module: `composer update && composer require --dev codeception/codeception weirdan/codeception-psalm-module`
2. Configure codeception it - use [existing working configuration](https://github.com/weirdan/doctrine-psalm-plugin/blob/master/codeception.yml), adjust the namespace
3. Configure the suite - use [this config](https://github.com/weirdan/doctrine-psalm-plugin/blob/master/tests/acceptance.suite.yml), put it into `tests/acceptance.suite.yml`, adjust namespaces
4. Generate helper: `vendor/bin/codecept generate:helper Acceptance`
5. Build the tester: `vendor/bin/codecept build`
6. Create directory for acceptance tests: `mkdir tests/acceptance`
7. Create your first test in `tests/acceptance/MyFeature.feature`:
    ```gherkin
    Feature: basics
      In order to test my plugin
      As a plugin developer
      I need to have tests

      Background:
        Given I have the following config
          """
          <?xml version="1.0"?>
          <psalm totallyTyped="true">
            <projectFiles>
              <directory name="."/>
            </projectFiles>
            <plugins>
              <pluginClass class="Weirdan\PsalmPluginSkeleton\Plugin" />
            </plugins>
          </psalm>
          """
      Scenario: run without errors
        Given I have the following code
          """
          <?php
          echo atan(1);
          """
        When I run Psalm
        Then I see no errors

      Scenario: run with errors
        Given I have the following code
          """
          <?php
          echo atan("not a number");
          """
        When I run Psalm
        Then I see these errors
          | Type                  | Message                                                         |
          | InvalidScalarArgument | Argument 1 of atan expects float, string(not a number) provided |
        And I see no other errors
    ```
 8. Run your tests: `vendor/bin/codecept run acceptance`:
    ![image](https://user-images.githubusercontent.com/57403/93808507-c1f91380-fc54-11ea-816a-8b257e8f8bbe.png)


Learn by example by reading [doctrine-psalm-plugin tests](https://github.com/weirdan/doctrine-psalm-plugin/tree/master/tests/acceptance), [psalm module own tests](https://github.com/weirdan/codeception-psalm-module/blob/master/tests/acceptance/PsalmModule.feature) and modules [release notes](https://github.com/weirdan/codeception-psalm-module/releases). If the syntax looks unfamiliar, it's [gherkin](https://cucumber.io/docs/gherkin/reference/), same as used by [Behat](https://docs.behat.org/en/latest/).

This actually doesn't have to be this complex, I'll look into adding most preparational steps to the skeleton itself.

**Comment by weirdan** [comment]
Fixed in #6
